core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -53,6 +53,8 @@ object RequestChannel extends Logging {
     @volatile var apiLocalCompleteTimeMs = -1L
     @volatile var responseCompleteTimeMs = -1L
     @volatile var responseDequeueTimeMs = -1L
+    @volatile var apiRemoteCompleteTimeMs = -1L
```

Comment:
@junrao : I looked again. The reason `apiLocalCompleteTimeMs` can be negative is that we update it in the `finally` clause of the `KafkaApis.handle` (which is the request handler thread). If the network thread gets to read it before that update (which is possible) then we can detect that by checking if it is negative and round up the `apiLocalCompleteTimeMs` to the `responseCompleteTimeMs`. @auradkar  can you add comments to `RequestChannel` for these: i.e., why it needs to be `volatile` and why `apiLocalCompleteTimeMs` can be negative?